modules/currency-l1/src/test/scala/org/tessellation/currency/l1/DummyDataApplicationL1Service.scala

```diff
@@ -20,7 +20,7 @@ class DummyDataApplicationL1Service extends BaseDataApplicationL1Service[IO] {
 
   override def deserializeState(bytes: Array[Byte]): IO[Either[Throwable, DataOnChainState]] = ???
 
-  override def serializeUpdate(update: DataUpdate): IO[Array[Byte]] = IO.delay(JsonBinarySerializer.serialize(update)(dataEncoder))
+  override def serializeUpdate(update: DataUpdate): IO[Array[Byte]] = IO.pure(JsonBinarySerializer.serialize(update)(dataEncoder))
```

Comment:
It is not a pure operation, it should be delayed.

Reply:
Updating